documentation/intro-to-ansible-for-system-roles.md

```diff
@@ -0,0 +1,120 @@
+META: ran handlers
+
+PLAY RECAP ******************************************************************************
+localhost                  : ok=2    changed=1    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0
+```
+For more details, visit this page:
+["Ansible playbooks"](https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_intro.html)
+
+## How to Create an Inventory
+
+To run Ansible, you have to define the `managed nodes` on which you want to
+install systems and configure them. The inventory or inventory file is used
+for this purpose. In the above "Simple playbook example," the playbook is
+executed on `localhost`, which is the same as the host where you run `
+ansible-playbook`. The host is called a `control node`.
+
+There are multiple ways to specify the `managed nodes`.
+You can embed it in the playbook. Or you can specify it in the `ansible-playbook`
+option directly or in the inventory file.
+```
+  -i INVENTORY, --inventory INVENTORY
+          specify inventory host path or comma separated host list.
+```
+For more details about inventory, you can start here:
+["How to build your inventory"](https://docs.ansible.com/ansible/latest/inventory_guide/intro_inventory.html)
+
+## How to Create a Playbook
+
+As presented in the "How to run Ansible" section, this Ansible documentation
+["Ansible playbooks"](https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_intro.html)
+is a good place to start.
+
+To use Linux System Roles from your playbook, you can include them in a legacy
+style, as described in
+["Roles"](https://docs.ansible.com/ansible/latest/playbook_guide/playbooks_reuse_roles.html)
+or in the collection format, as described in
+["Using collections in playbooks"](https://docs.ansible.com/ansible/latest/collections_guide/collections_using_playbooks.html#using-collections-in-playbooks)
+The Linux System Roles support both styles.
+
+## How to Use Vault
+
+You may need to define parameters such as passwords and other secrets
+in your playbook. Some of the Linux System Roles require such sensitive
+parameters to configure them in the supported system.
+If you use the `ansible-vault` tool included in the Ansible package,
+they are encrypted and stored safely in your playbook.
+
+The following command-line encrypts a secret value "my_secret_value"
+using the vault password stored in /path/to/vault_pwd and stores the encrypted
+secret in /path/to/vault-variables.txt:
+```
+ansible-vault encrypt_string --vault-password-file /path/to/vault_pwd \
+my_secret_value --name my_secret_var_name > /path/to/vault-variables.txt
+```
+
+The generated /path/to/vault-variables.txt file looks like this:
+```
+my_secret_var_name: !vault |
+          $ANSIBLE_VAULT;1.1;AES256
+          62393438633534323463643838313865386265363962313861303564383137653862363237323332
+          <<snip>>
+```
+Instead of specifying the real secret value "my_secret_value", you can use
+"my_secret_var_name" in your playbook.
```

Comment:
```yaml
- name: Use the secret
  some_task:
    password: "{{ my_secret_var_name }}"
  ...
  no_log: true
```
Ansible will keep the value of `my_secret_var_name` encrypted until needed (and use `no_log: true` to avoid leaking the value in the Ansible logs).